### PR TITLE
Access/Airlock Revamp

### PR DESCRIPTION
**Pull request by IamVelcroboy**
## About the PR
Revamps access for several roles:
- ~~Changes Brig access to Lawyer~~
- Adds Sec/Lawyer airlocks ~~in place of brig~~
- Adds airlocks for Lawyer that are not Security colored
- ~~Gives Reporter, Librarian, and Zookeeper their own accesses and airlocks~~
- Adds a hydroponics-locked freezer door
- Adds a glass airlock for Chemistry
- Gives all heads a maintenance airlock
- Adds a service maintenance airlock
- Gives ~~Chaplain,~~ Janitor, ~~Clown, Mime, and musician Service access~~
- ~~Gives Boxer and Reporter Theatre(entertainment) access~~

## Why / Balance
- Biggest issue was Brig/Lawyer being confusing. Lawyer should simply have their own access.
- Hydro-locked access was missing for shared freezer rooms
- Chem glass airlock was missing
- Service roles should have access to service-locked rooms for service-based lounge/equipment rooms
- Boxer and Reporter fall under the "entertainment" category, thus should have Theatre access
- Reporters should get their own access so randos aren't using their equipment/terminals
- Zookeeper should get their own access to keep their animals secured
- Librarian should have their own access so they can have a private office 
- Head and service maint airlocks were missing. Adding them will allow mappers to add access to maints to/from their offices

## Technical details
n/a

## Media
Not really necessary unless you want a bunch of pictures of airlocks.
- [x] I have added screenshots/videos to this PR showcasing its changes in-game, **or** this PR does not require an in-game showcase

## Breaking changes
Nothing should break as migration was set up for the brig/lawyer change. However, mappers should make adjustments as soon as they can.

**Changelog**
n/a Mostly mapping related

### CODE REVIEW

**Comment by lzk228** [comment]
i don't like that brig access removed :cry:

**Comment by IamVelcroboy** [comment]
> i don't like that brig access removed 😢

What is it needed for? It's only used to give lawyers access to certain areas of sec. A combination sec/lawyer access door was added to facilitate this instead.

**Comment by deltanedas** [comment]
brig is just lawyer access, both lawyer and sec have it

File: Resources/migration.yml [hunk 1]
```diff
@@ -87,3 +87,8 @@ GeneratorPlasma: PortableGeneratorPacman
 GeneratorUranium: PortableGeneratorSuperPacman
 GeneratorPlasmaMachineCircuitboard: PortableGeneratorPacmanMachineCircuitboard
 GeneratorUraniumMachineCircuitboard: PortableGeneratorSuperPacmanMachineCircuitboard
+
+# 2023-11-16
+AirlockBrigLocked: AirlockSecurityLawyerLocked
+AirlockBrigGlassLocked: AirlockSecurityLawyerGlassLocked
+WindoorSecureBrigLocked: WindoorSecureLawyerLocked
```
Comment (lzk228) [inline Resources/migration.yml:94]: isn't there should be WindoorSecureSecurityLawyerLocked?

**Review by lzk228 — CHANGES_REQUESTED** [review]
boxer and reporter aren't the part of theatre

also change accesses in `Content.Shared/Access/Components/SharedIdCardConsoleComponent.cs`

**Comment by IamVelcroboy** [comment]
> also change accesses in `Content.Shared/Access/Components/SharedIdCardConsoleComponent.cs`

Thanks! I was trying to locate that.

**Comment by IamVelcroboy** [comment]
> boxer and reporter aren't the part of theatre

But they should be. It all falls under entertainment.

File: Resources/migration.yml (hunk 1 above)
Comment (IamVelcroboy) [reply Resources/migration.yml:94]: Good catch, yes

**Comment by LankLTE** [comment]
I don’t feel reporter, zookeeper, and *especially* librarian need their own access. Given paramedics were previously stated not to need access I don’t think these RP roles need one either, it just kind of keeps people out of random rooms that them being in isn’t an issue anyways. Could you explain why the librarian having their own locked room is important enough to warrant having its’ own access?

**Comment by IamVelcroboy** [comment]
> Could you explain why the librarian having their own locked room is important enough to warrant having its’ own access?

Privacy, giving them an office/bedroom that anyone can walk into is weird.

**Comment by IamVelcroboy** [comment]
Reporter and zookeeper certainly have reasons for unique access as I mentioned in the description.

**Comment by 0x6273** [comment]
> Gives Chaplain, Janitor, Clown, Mime, and musician Service access

Why does clown and mime need service access?

**Comment by LordEclipse** [comment]
Brig access is used for the brig, where you place people that were arrested. Lawyer is supposed to be able to access the brig. It's a bit weird to be labeling the brig with "Lawyer" access doors. Also access for niche roles is a bit much imo.

**Comment by IamVelcroboy** [comment]
> Why does clown and mime need service access?

Because they are a part of the Service dept. And we map service rooms in our maps that they should have access to.

**Comment by IamVelcroboy** [comment]
> Brig access is used for the brig, where you place people that were arrested. Lawyer is supposed to be able to access the brig. It's a bit weird to be labeling the brig with "Lawyer" access doors. Also access for niche roles is a bit much imo.

But that's not actually how they are used. It's confusing and mislabeled.

Brig is only used in the entry to sec dept. Not on the jail cells or perma as it would indicate. Sec access is used for those. Brig is used to give lawyers access to certain areas. So why not call it what it is and stop confusing new mappers?

**Comment by IamVelcroboy** [comment]
> Brig access is used for the entry, as well as all jail cells, it is not used for perma.

It is not used for jail cells. That would allow lawyers to let people out of them.

**Comment by LordEclipse** [comment]
> > Brig access is used for the entry, as well as all jail cells, it is not used for perma.
> 
> It is not used for jail cells. That would allow lawyers to let people out of them.

No, its always used for jail cells.

**Comment by deltanedas** [comment]
no it is not. go check any map :trollface:

**Comment by IamVelcroboy** [comment]
This is exactly the type of confusion I'm trying to fix...

**Comment by LordEclipse** [comment]
> no it is not. go check any map :trollface:

Huh, it used  to be that way.

**Comment by Emisse** [comment]
I disagree with large portions of this pr, could you atomize this into smaller chunks

**Review by Emisse — CHANGES_REQUESTED** [review]
Atomize PR

**Comment by IamVelcroboy** [comment]
> I disagree with large portions of this pr, could you atomize this into smaller chunks

Could you specify which parts you disagree with?

**Comment by Emisse** [comment]
Changes Brig access to Lawyer
Adds Sec/Lawyer airlocks in place of brig
Gives Reporter, Librarian, and Zookeeper their own accesses and airlocks
Gives Chaplain, Janitor, Clown, Mime, and musician Service access
Gives Boxer and Reporter Theatre(entertainment) access

**Comment by deltanedas** [comment]
> Changes Brig access to Lawyer
Adds Sec/Lawyer airlocks in place of brig

these are good since it prevents this confusion:

> > > Brig access is used for the entry, as well as all jail cells, it is not used for perma.

> > It is not used for jail cells. That would allow lawyers to let people out of them.

> No, its always used for jail cells.


others i agree they are excessive

**Comment by Emisse** [comment]
id be more fine with it if it was a single PR, needs atomization

**Comment by IamVelcroboy** [comment]
Brig access is confusing and not even used for what it implies. All it really is is lawyer access. Why not call it what it is? 

Lawyers shouldn't have security doors for their offices. They are not sec. 

The rest of the changes offer better granularity for mappers to add more nuance to their maps.

Most of these roles should have areas where not anyone and everyone can just walk in. Librarian is probably the only one that's a bit of a stretch, but then why does chaplain get a unique access? 

Why would you not want service roles to have service access? Especially janitor?

Why would you not want at least boxer to have theater access? They are a part of the entertainment crew. Reporter is basically entertainment as well.

**Comment by IamVelcroboy** [comment]
The problem with breaking this up is they all touch the same files. It would be a headache with merge conflicts on every individual PR.

I'd really rather just adjust this one to what you'll accept and I'll take what don't downstream where we do want these changes. I just though it made more sense to start at the top so it would trickle down to forks.

**Comment by Emisse** [comment]
Keep brig access the same, it is for access to brig, make a new door for lawyer for law office if you want.
Reporter Librarian and zookeeper don't need unique accesses. Chaplain only has a unique access because of crematorium.
Janitor is fine to have service access, the rest aren't.
Boxer and Reporter shouldn't have entertainment access.

**Comment by deltanedas** [comment]
> why does chaplain get a unique access?

little incincerator device in chaps room :trollface:

**Comment by IamVelcroboy** [comment]
If everyone in the sec department has brig access, why not just use sec access? Unless there are plans to limit who has access to the brig? I'm not aware of any discussion to do that. I just don't see how brig is needed? 

Reporter has cameras that people can walk off with and the mass media console that they should have exclusive access to.

Zookeeper should be able to keep their animals secured.

**Comment by Emisse** [comment]
Brig lets people into main cell block of security, people can have brig but not be able to get into locker rooms or other rooms in sec.

**Comment by IamVelcroboy** [comment]
> Brig lets people into main cell block of security, people can have brig but not be able to get into locker rooms or other rooms in sec.

The only person that applies to us lawyer, which is why sec/lawyer access makes more sense

**Comment by LankLTE** [comment]
You don’t need individual access for every role in the game. If people are walking off with your animals, or stealing your reporter’s cameras, that’s an IC problem and can be dealt with IC rather than being prevented through new accesses that aren’t even relevant on every station.

**Comment by IamVelcroboy** [comment]
> You don’t need individual access for every role in the game. If people are walking off with your animals, or stealing your reporter’s cameras, that’s an IC problem and can be dealt with IC rather than being prevented through new accesses that aren’t even relevant on every station.

I disagree. Granularity of access is much better.

**Comment by deltanedas** [comment]
nah fuck zookeepers chef can have their animals :trollface::trollface::trollface:

File: Resources/Locale/en-US/prototypes/access/accesses.ftl [hunk 2]
```diff
@@ -24,16 +24,14 @@ id-card-access-level-cargo = Cargo
 id-card-access-level-salvage = Salvage
 
 id-card-access-level-bar = Bar
+id-card-access-level-bar = Brig
```
[the remark this answers is not in the document]
Comment (IamVelcroboy) [reply Resources/Locale/en-US/prototypes/access/accesses.ftl:27]: oops

**Comment by IamVelcroboy** [comment]
> Brig lets people into main cell block of security, people can have brig but not be able to get into locker rooms or other rooms in sec.

Ok requested changes were made.
- Redundant brig access was re-added
- Service was removed from all but janitor
- Reporter, Library, and Zookeeper access and airlocks were removed
- Theater was removed from Boxer and Reporter 
- Reverted migration

**Pull request merged**